go-controller/pkg/node/node.go

```diff
@@ -594,16 +594,25 @@ func (n *OvnNode) WatchEndpoints() {
 	}, nil)
 }
 
-// validateGatewayMTU checks if the MTU of the given network interface is big
-// enough to carry the `config.Default.MTU` and the Geneve header. If the MTU
-// is not big enough, it will taint the node with the value of
-// `types.OvnK8sSmallMTUTaintKey`
-func (n *OvnNode) validateGatewayMTU(gatewayInterfaceName string) error {
+// validateVTEPInterfaceMTU checks if the MTU of the interface that has ovn-encap-ip is big
+// enough to carry the `config.Default.MTU` and the Geneve header. If the MTU is not big
+// enough, it will taint the node with the value of `types.OvnK8sSmallMTUTaintKey`
+func (n *OvnNode) validateVTEPInterfaceMTU() error {
 	tooSmallMTUTaint := &kapi.Taint{Key: types.OvnK8sSmallMTUTaintKey, Effect: kapi.TaintEffectNoSchedule}
 
-	mtu, err := util.GetNetworkInterfaceMTU(gatewayInterfaceName)
+	ovnEncapIPStr, stderr, err := util.RunOVSVsctl("--if-exists", "get",
```

Comment:
I don't think we need to read the value from OVS and parse the string again into an IP. Why dont we just set the config value rather than doing it in this function:

```
func setupOVNNode(node *kapi.Node) error {
	var err error

	encapIP := config.Default.EncapIP
	if encapIP == "" {
		encapIP, err = util.GetNodePrimaryIP(node)
		if err != nil {
			return fmt.Errorf("failed to obtain local IP from node %q: %v", node.Name, err)
		}
	} else {
		if ip := net.ParseIP(encapIP); ip == nil {
			return fmt.Errorf("invalid encapsulation IP provided %q", encapIP)
		}
	}
```

Reply:
@trozet I can't do this in pkg/config/config.go since we need to call into  util.GetNodePrimaryIP(node) and I don't have `node *kapi.Node` in that function.

Reply:
@trozet I have addressed your comments. PTAL